catalog/catalog-client/src/main/java/com/cgdpd/library/catalog/client/rest/LibraryCatalogHttpClient.java

```diff
@@ -27,4 +30,19 @@ public Mono<DetailedBookDto> getDetailedBookDto(Isbn13 isbn13) {
               .bodyToMono(DetailedBookDto.class)
               .onErrorMap(onErrorMap());
     }
+
+    @Override
+    public Flux<DetailedBookDto> searchBooks(PaginationCriteria paginationCriteria,
```

Comment:
Can't we do this instead?
```suggestion
    public Mono<PagedResponse<DetailedBookDto>> searchBooks(PaginationCriteria paginationCriteria,
```
Also change name to `getBooksByCriteria`